cpt/cpp_sources/Scorer.cpp

```diff
@@ -30,3 +31,28 @@ int Scorer::get_best_prediction() const
 {
     return std::distance(m_data.begin(), std::max_element(m_data.begin(), m_data.end()));
 }
+
+std::vector<int> Scorer::get_best_k_predictions(std::size_t k) const
```

Comment:
Good but this should use `std::partial_sort_copy`.

Reply:
I should probably at least do some benchmarks. I am just afraid that the full copy of the `m_data` costs a bit too much.

I will do benchmarks, and depending on them I will choose (I prefer `partial_sort_copy` for readability)